keras_nlp/models/t5/t5_backbone.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""T5 backbone model."""
+
+import copy
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.layers.transformer_layer_utils import compute_causal_mask
+from keras_nlp.models.backbone import Backbone
+from keras_nlp.models.t5.t5_layer_norm import T5LayerNorm
+from keras_nlp.models.t5.t5_presets import backbone_presets
+from keras_nlp.models.t5.t5_transformer_layer import T5TransformerLayer
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras_nlp_export("keras_nlp.models.T5Backbone")
+class T5Backbone(Backbone):
+    """T5 encoder-decoder backbone model.
+
+    T5 is a LLM pretrained on a mix of unsupervised and supervised tasks,
+    where each task is converted to a sequence-to-sequence format.
+    T5 works well on a variety of tasks out-of-the-box by prepending
+    various prefixex to the input sequence, e.g., for translation:
+    `"translate English to German: ..."`, for summarization:
+    `"summarize: ..."`.
+
+    T5 was introduced in
+    [Exploring the Limits of Transfer Learning with a Unified Text-to-Text Transformer](https://arxiv.org/abs/1910.10683)
+
+    The default constructor gives a fully customizable, randomly initialized T5
+    model with any number of layers, heads, and embedding dimensions. To load
+    preset architectures and weights, use the `from_preset` constructor.
+
+    Disclaimer: Pre-trained models are provided on an "as is" basis, without
+    warranties or conditions of any kind. The underlying model is provided by
```

Comment:
I think we decided that for google models we don't need to link out (as long as mit or apache). Just stop at the first sentence.

Reply:
Done.